### PR TITLE
feat(lib-shared-advert-card):[skp-69] create advert card

### PR DESCRIPTION
**Pull request by MariuzM**
Please review my PR

Known Issue: 
- When compiling to storybook getting this error 
Screen Shot 2020-06-11 at 20 15 47

I have two interfaces and looks like only one being passed but not the other one `Author` 
```js
export interface SharedAdvertCardProps {
  title: string;
  author: Author;
  timeStamp: number;
  description: string;
  like: number;
  shared: number;
  imgUrl: string;
}

export interface Author {
  _id: number;
  firstName: string;
  lastName: string;
  img: string;
}
```

### CODE REVIEW

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        <img src={props.imgUrl} alt="" />
+      </div>
+
+      <div className="social-wrapper flex-wrap">
+        <button
+          type="button"
+          className="wrapper-like flex-wrap"
+          onClick={onLikeClick}
+        >
+          <IconHeart />
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:168]: use https://material-ui.com/components/material-icons/ icons, same below

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            alt=""
+            height="40"
+            width="40"
+          />
+        </button>
+
+        <div className="card-title">
+          <a
+            href="url"
+            onClick={(event): void => onViewClick(event, props.author._id)}
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:143]: () => onViewClick(props.author._id)

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  const onSharedClick = (): void => {
+    // console.log('onSharedClick');
+  };
+
+  return (
+    <div className="card">
+      <div className="card-title-wrap flex-wrap">
+        <button
+          type="button"
+          onClick={(event): void => onViewClick(event, props.author._id)}
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:129]: no need event () => onViewClick(props.author._id)

File: libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { text, , number, text, number, number, text } from '@storybook/addon-knobs';
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx:1]: remove duplicated imports

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        aria-haspopup="true"
+        onClick={handleClick}
+      >
+        <MoreVertIcon />
+      </IconButton>
+
+      <Menu
+        id="long-menu"
+        anchorEl={anchorEl}
+        keepMounted
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:39]: remove

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+  title: string;
+  author: Author;
+  timeStamp: number;
+  description: string;
+  like: number;
+  shared: number;
+  imgUrl: string;
+}
+
+export interface Author {
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:100]: author interface should be upper then SharedAdvertCardProps

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+            href="url"
+            onClick={(event): void => onViewClick(event, props.author._id)}
+          >
+            <span style={{ fontWeight: 'bold' }}>
+              {`${props.author.firstName} ${props.author.lastName} `}
+            </span>
+          </a>
+
+          <span>posted on your timeline</span>
+          <p style={{ color: '#8B8B8B' }}>{`${props.timeStamp} Minutes ago`}</p>
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:151]: use style file for styles

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 8]
```diff
@@ -0,0 +1,187 @@
+import React from 'react';
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:1]: import React, {useState} from 'react';

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+export interface SharedAdvertCardProps {
+  title: string;
+  author: Author;
+  timeStamp: number;
+  description: string;
+  like: number;
+  shared: number;
+  imgUrl: string;
+}
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:98]: events add in this interface ex.: onViewClick: (id: string) => void

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 10]
```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+
+import './shared-advert-card.scss';
+
+const MenuButton = () => {
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:9]: this should be as external component

File: libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  component: SharedAdvertCard,
+  title: 'SharedAdvertCard'
+};
+
+export const primary = () => {
+  
+  
+  const props:SharedAdvertCardProps = {
+    title: text('title', ''),
+    author: ('author', ),
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx:15]: author: text('author', ''),

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+      width="24"
+      height="24"
+      viewBox="0 0 24 24"
+    >
+      <path d="M12 4.435c-1.989-5.399-12-4.597-12 3.568 0 4.068 3.06 9.481 12 14.997 8.94-5.516 12-10.929 12-14.997 0-8.118-10-8.999-12-3.568z" />
+    </svg>
+  );
+};
+
+const IconShare = () => {
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:76]: remove

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+            {option}
+          </MenuItem>
+        ))}
+      </Menu>
+    </div>
+  );
+};
+
+const IconHeart = () => {
+  return (
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:64]: remove

File: libs/shared/advert-card/src/lib/shared-advert-card.scss [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+        span {
+          font-weight: bold;
+        }
+      }
+      margin-left: 10px;
+      a {
+        color: #4b89c4;
+      }
+    }
+    // div:last-child {
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.scss:45]: remove if unnecessary

File: libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,32 @@
+import { number, text } from '@storybook/addon-knobs';
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx:1]: + object

File: libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+
+export default {
+  component: SharedAdvertCard,
+  title: 'SharedAdvertCard',
+};
+
+export const primary = () => {
+  const props: SharedAdvertCardProps = {
+    title: text('title', ''),
+    author: text('author', ''),
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx:13]: text('author', '') => object('author', {_id: '', firstName: '', lastName: '', img: ''})

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 17]
```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import FavoriteIcon from '@material-ui/icons/Favorite';
+import ShareIcon from '@material-ui/icons/Share';
+// import { SharedMenuButton } from '@ppm/shared/menu-button';
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:4]: remove if unnecessary

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+);
+
+export const SharedAdvertCard = (props: SharedAdvertCardProps) => {
+  const [timeState, setTimeState] = useState('');
+  const classes = useStyles();
+
+  function timeCalculator() {
+    const dbDate = '2020-06-13 18:33:30';
+    // const yyyy: number = +dbDate.slice(0, 4);
+    // const mm: number = +dbDate.slice(5, 7);
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:43]: remove all commented code

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+            height="40"
+            width="40"
+          />
+        </button>
+
+        <div className="card-title">
+          <a href="url" onClick={() => props.onViewClick(props.author._id)}>
+            <span>
+              {`
+								${props.author.firstName || ''} ${props.author.lastName || ''}
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:107]: bad format

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx [hunk 20]
```diff
@@ -0,0 +1,148 @@
...
+								${props.author.firstName || ''} ${props.author.lastName || ''} 
+							`}
+            </span>
+          </a>
+
+          <span>posted on {props.title}</span>
+          <p>{`${timeState} ago`}</p>
+        </div>
+
+        {/* <SharedMenuButton /> */}
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.tsx:116]: remove

File: libs/shared/advert-card/src/lib/shared-advert-card.tsx (hunk 19 above)
Comment (MariuzM) [reply libs/shared/advert-card/src/lib/shared-advert-card.tsx:99]: Looks fine for me, i assume that this is issue with Github not formatting white spaces / tab properly like I've done in VSCode? 
image

File: libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx (hunk 16 above)
Comment (MariuzM) [reply libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx:13]: Added but getting errors:
image

File: libs/shared/advert-card/src/lib/shared-advert-card.scss [hunk 21]
```diff
@@ -0,0 +1,67 @@
+.flex-wrap {
```
[the remark this answers is not in the document]
Comment (MariuzM) [reply libs/shared/advert-card/src/lib/shared-advert-card.scss]: Unsure about this, could you expand on this please

File: libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx (hunk 15 above)
Comment (MariuzM) [reply libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx:1]: Unsure about this, could you expand on this please

File: libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+export default {
+  component: SharedAdvertCard,
+  title: 'SharedAdvertCard',
+};
+
+export const primary = () => {
+  const props: SharedAdvertCardProps = {
+    title: text('title', ''),
+    author: object('author', { _id: '', fistName: '', lastName: '', img: '' }),
+    timeStamp: number('timeStamp', 0),
```
Comment (doncevas) [inline libs/shared/advert-card/src/lib/shared-advert-card.stories.tsx:14]: "number" change to "text"

**Pull request merged**